### PR TITLE
Add recipe for conda-devenv

### PR DESCRIPTION
**Pull request by nicoddemus**
cc @tadeu @gqmelo @edisongustavo

### CODE REVIEW

**Comment by nicoddemus** [comment]
AppVeyor is currently broken: conda-forge/conda-smithy#489

File: recipes/conda-devenv/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{% set name = "conda-devenv" %}
+{% set version = "0.9.0" %}
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_rev: {{ version }}
+  git_url: https://github.com/ESSS/conda-devenv.git
```
Comment (jakirkham) [inline recipes/conda-devenv/meta.yaml]: Is there a tag or release on PyPI that we can use?

File: recipes/conda-devenv/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
```
Comment (jakirkham) [inline recipes/conda-devenv/meta.yaml]: Please add...

```yaml
  entry_points:
    - conda-devenv = conda_devenv.devenv:main
```

**Comment by nicoddemus** [comment]
CI just passed. 👍

**Comment by nicoddemus** [comment]
Oops @jakirkham did not see your comments.

File: recipes/conda-devenv/meta.yaml (hunk 1 above)
Comment (nicoddemus) [inline recipes/conda-devenv/meta.yaml]: There's a tag on the repository, but we don't pretend to release to PyPI since it is meant to be used with conda only (like `conda-smithy` itself for example).

File: recipes/conda-devenv/meta.yaml (hunk 2 above)
Comment (nicoddemus) [inline recipes/conda-devenv/meta.yaml]: Roger

File: recipes/conda-devenv/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/conda-devenv/meta.yaml]: Tag on the repo is fine. We just prefer to download a tarball instead of cloning is all.

File: recipes/conda-devenv/meta.yaml (hunk 2 above)
Comment (jakirkham) [inline recipes/conda-devenv/meta.yaml]: note: Done below.

**Comment by nicoddemus** [comment]
Hmm it seems we have reached the rate limit for AppVeyor.

**Comment by MSeifert04** [comment]
There's a rate limit?

File: recipes/conda-devenv/meta.yaml (hunk 1 above)
Comment (nicoddemus) [inline recipes/conda-devenv/meta.yaml]: Oh OK, will update it.

**Comment by jakirkham** [comment]
Yes, there is a [GitHub rate limit]( https://developer.github.com/v3/#rate-limiting ) per user for requests to GitHub. Basically a security measure to avoid flooding GitHub with requests, but also some other reasons as well. AppVeyor makes all calls with one user, but I think other CIs attach them to the user doing the PR. Not sure why they can't change it though. 😕 ( https://github.com/appveyor/ci/issues/1002 )

**Comment by jakirkham** [comment]
Well that rate limit ended really fast. We must have been down to the last couple minutes. Take the win I guess. 😄

File: recipes/conda-devenv/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+{% set name = "conda-devenv" %}
+{% set version = "0.9.1" %}
+{% set hash = "fe8e9be7d4f2df03b9a269a4b4b36ca67477c9b1d5c2af1af440f63b787f2f05" %}
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: '{{ version }}.tar.gz'
```
Comment (jakirkham) [inline recipes/conda-devenv/meta.yaml]: Optional, but could structure your filename like [this]( https://github.com/conda-forge/staged-recipes/blob/0394a09dfc8024382222d1cb77a523d7647734e3/recipes/example/meta.yaml#L17 ). Helps with caching (and finding things 😄) if one does a lot of builds locally.

File: recipes/conda-devenv/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+{% set name = "conda-devenv" %}
+{% set version = "0.9.1" %}
+{% set hash = "fe8e9be7d4f2df03b9a269a4b4b36ca67477c9b1d5c2af1af440f63b787f2f05" %}
+package:
```
Comment (jakirkham) [inline recipes/conda-devenv/meta.yaml]: Could we please have a newline between `package` and the Jinja?

File: recipes/conda-devenv/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+source:
+  fn: '{{ version }}.tar.gz'
+  url: https://github.com/ESSS/conda-devenv/archive/{{ version }}.tar.gz
+  sha256: {{ hash }}
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+  entry_points:
+      - conda-devenv = conda_devenv.devenv:main
```
Comment (jakirkham) [inline recipes/conda-devenv/meta.yaml]: nit: Would be nice if the indent size is uniform (e.g. only 2 spaces more here than `entry_points`).

File: recipes/conda-devenv/meta.yaml (hunk 4 above)
Comment (nicoddemus) [inline recipes/conda-devenv/meta.yaml]: Sure done

File: recipes/conda-devenv/meta.yaml (hunk 5 above)
Comment (nicoddemus) [inline recipes/conda-devenv/meta.yaml]: Oops sure

**Comment by nicoddemus** [comment]
Hmm I just noticed recently that the Travis builds are starting much quicker than before. Did Travis add more OSX hosts to its build farm perhaps?

**Comment by jakirkham** [comment]
I don't think they have added any workers.

The real cause, I believe, was fixing this [issue]( https://www.traviscistatus.com/incidents/ld48m3mfd0g3 ). Basically we started noticing that simple steps were taking absurd amounts of time (particularly at high usage times). This likely caused this massive build in the queue as workers took longer to do the same thing than usual.

Shortly after we [pointed this out]( https://github.com/travis-ci/travis-ci/issues/7190#issuecomment-286156602 ) to them, that issue got opened and the problem resolved. They also did some follow-up [maintenance]( https://www.traviscistatus.com/incidents/w5zv3135z4dx ). After they completed the maintenance, we start seeing a significant improvement (see a screenshot of the status below).

In other words, we have likely been suffering from degraded service for months, which has caused the massive backlogs. Note the backlogs were in the ~1000s before now they are in the ~100s. So order of magnitude improvement.

FWIW you can thank @gqmelo for helping discover this issue.

screen shot 2017-03-23 at 19 52 20

**Comment by nicoddemus** [comment]
Nice @jakirkham, thanks for the detailed response! I will thank @gqmelo personally tomorrow then! 👍

**Comment by jakirkham** [comment]
Thanks @nicoddemus.
